src/idl_gen_cpp.cpp

```diff
@@ -1535,6 +1535,40 @@ class CppGenerator : public BaseGenerator {
     }
   }
 
+  // Generate CompareWithValue method for a key field.
+  void GenKeyFieldMethods(const FieldDef &field) {
+    FLATBUFFERS_ASSERT(field.key);
+    const bool is_string = (field.value.type.base_type == BASE_TYPE_STRING);
+
+    code_ += "  bool KeyCompareLessThan(const {{STRUCT_NAME}} *o) const {";
+    if (is_string)
+    { // use operator< of flatbuffers::String
+      code_ += "    return *{{FIELD_NAME}}() < *o->{{FIELD_NAME}}();";
+    } else {
+      code_ += "    return {{FIELD_NAME}}() < o->{{FIELD_NAME}}();";
+    }
+    code_ += "  }";
+
+    if (is_string) {
```

Comment:
Thanks for the refactor! Even though `is_string` will never be true for structs, it is cleaner this way. And nice to use the trick with `-` for both.